crates/primitives/src/result.rs

```diff
@@ -122,25 +122,26 @@ impl Output {
 
 #[derive(Debug, Copy, Clone, PartialEq, Eq)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[non_exhaustive]
 pub enum EVMError<DBError> {
     Transaction(InvalidTransaction),
-    /// REVM specific and related to environment.
+    /// `prevrandao` is not set for Merge and above.
     PrevrandaoNotSet,
+    /// `excess_blob_gas` is not set for Cancun and above.
+    ExcessBlobGasNotSet,
```

Comment:
Would be good to have another type of errors `EnvError` so we would have `Transaction`, `Database` and `EnvError` types, and move `PrevrandaoNotSet` and `ExcessBlobGasNotSet` there.

Can be done after this PR, will make a issue.

Reply:
That's fine